framework/source/class/qx/bom/IdleCallback.js

```diff
@@ -0,0 +1,109 @@
+/* ************************************************************************
+
+   qooxdoo - the new era of web development
+
+   http://qooxdoo.org
+
+   Copyright:
+     2016 GONICUS GmbH, Germany, http://www.gonicus.de
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * Cajus Pollmeier (cajus)
+
+************************************************************************ */
+
+/**
+ * This is a cross browser wrapper for requestIdleCallback. For further
+ * information about the feature, take a look at spec:
+ * https://www.w3.org/TR/requestidlecallback/
+ *
+ * Here is a sample usage:
+ * <pre class='javascript'>var clb = function(deadline) {
+ *   while (deadline.timeRemaining() > 0) {
+ *     // ... do some last tasks
+ *   }
+ * };
+ *
+ * qx.bom.IdleCallback.request(clb, this);
+ * </pre>
+ */
+qx.Bootstrap.define("qx.bom.IdleCallback",
+{
+  extend : qx.core.Object,
+
+  statics :
+  {
+    /**
+     * The default time in ms the timeout fallback implementation uses.
+     */
+    TIMEOUT : 0,
+
+    /**
+     * The default remaining time in ms the timeout fallback implementation uses.
+     */
+    REMAINING : 250,
+
+    /**
+     * Request for an IDLE callback. If the native <code>requestIdleCallback</code>
+     * method is supported, it will be used. Otherwise, we use timeouts with a
+     * 30ms delay. The HighResolutionTime will be used if supported but the time given
+     * to the callback will still be a timestamp starting at 1 January 1970 00:00:00 UTC.
+     *
+     * @param callback {Function} The callback function which will get a deadline
+     *   object. It contains a <code>timeRemaining()</code> call which returns the
+     *   remaining milliseconds and the <code>didTimeout</code> flag which indicates
+     *   whether the callback was fired due to a timeout. The latter is always false
+     *   in case of the emulation.
+     * @param context {var} The context of the callback.
+     * @param timeout {Number} Timeout in milliseconds.
+     * @return {Number} Handle for that request
+     */
+    request : function(callback, context, timeout) {
+      var clb = function(deadline) {
```

Comment:
Since you need a name other than `callback`, the most commonly-seen abbreviation for it is `cb`. I had to spend some time figuring out what `clb` meant. Suggest changing it to `cb`.